frontend/e2e/amending-law-metadata-editor-rahmen.spec.ts

```diff
@@ -0,0 +1,842 @@
+    ).toBeVisible()
+
+    await page.unrouteAll()
+  })
+})
+
+test.describe("metadata view", () => {
+  let sharedPage: Page
+
+  async function restoreInitialState() {
+    const data: Proprietary = {
+      fna: "210-5",
+      art: "regelungstext",
+      typ: "gesetz",
+      subtyp: "Rechtsverordnung",
+      bezeichnungInVorlage: "Testbezeichnung nach meiner Vorlage",
+      artDerNorm: "SN,ÜN",
+      normgeber: "BEO - Berlin (Ost)",
+      beschliessendesOrgan: "BMinJ - Bundesministerium der Justiz",
+      qualifizierteMehrheit: true,
+      federfuehrung: "BMI - Bundesministerium des Innern und für Heimat",
+    }
+
+    await sharedPage.request.put(
+      "/api/v1/norms/eli/bund/bgbl-1/1990/s2954/2023-12-29/1/deu/regelungstext-1/proprietary/2023-12-30",
+      { data },
+    )
+  }
+
+  async function saveMetadata() {
+    await sharedPage
+      .getByRole("button", { name: "Metadaten speichern" })
+      .click()
+    await sharedPage.waitForResponse(/proprietary/)
+  }
+
+  async function mockPutResponse(data: Proprietary) {
+    await sharedPage.route(/\/proprietary\/2023-12-30/, async (route) => {
+      if (route.request().method() === "PUT") {
+        const response = await route.fetch()
+        const body = await response.json()
+
+        await route.fulfill({
+          response,
+          body: JSON.stringify({ ...body, ...data }),
+        })
+      } else await route.continue()
+    })
+  }
+
+  test.beforeAll(async ({ browser }) => {
+    sharedPage = await browser.newPage()
+
+    await restoreInitialState()
+
+    await sharedPage.goto(
+      "/amending-laws/eli/bund/bgbl-1/2023/413/2023-12-29/1/deu/regelungstext-1/affected-documents/eli/bund/bgbl-1/1990/s2954/2023-12-29/1/deu/regelungstext-1/edit/2023-12-30",
+    )
+  })
+
+  test.afterAll(async () => {
+    await restoreInitialState()
+  })
+
+  test.describe("FNA", () => {
+    let control: Locator
+
+    test.beforeAll(() => {
+      control = sharedPage.getByRole("textbox", { name: "Sachgebiet" })
+    })
+
+    test("is loaded", async () => {
+      // Then
+      await expect(control).toHaveValue("210-5")
+    })
+
+    test("saves changes", async () => {
+      // When
+      await control.fill("123-4")
+      await saveMetadata()
+      await sharedPage.reload()
+
+      // Then
+      await expect(control).toHaveValue("123-4")
+    })
+
+    test("is updated with backend state after saving", async () => {
+      // Given
+      await mockPutResponse({ fna: "fake-backend-state" })
+      await expect(control).toHaveValue("123-4")
+
+      // When
+      await saveMetadata()
+
+      // Then
+      await expect(control).toHaveValue("fake-backend-state")
+
+      // Cleanup
+      await sharedPage.unrouteAll()
+    })
+  })
+
+  test.describe("Dokumenttyp", () => {
+    let control: Locator
+
+    test.beforeAll(() => {
+      control = sharedPage.getByRole("combobox", { name: "Dokumenttyp" })
+    })
```

Comment:
I'm a bit unsure about moving this up here, especially as it is only one line that is needed for the setup and one now needs to scroll around to understand what kind of element `control` is. I would have just duplicated it in all test cases.

Reply:
It is not always just one line ;) (see "art der norm") I proposed @andreasphil to move the control app to a `beforeAll` so that we don't need to duplicate it every time (always when using the string for the name you have the danger to misspell it somewhere).

But if it is about which kind of element is that, we could name the variables better for every metadatum, like for example `checkboxSN`, or `dropdownDocumentType`

I do understand the meaning of duplicating stuff for better readability of a test, but for these ones where we have a describe per metadatum and if we better-name the variables, I think the readability would still be given

Reply:
Thanks for the feedback, I think both are valid so let's leave that to whoever implements the test :) I improved the naming though to be more descriptive, the `control` was mostly for copy-pastability but you're right that it's not very helpful.